All_Relationships/3.model-label-functions.py

```diff
@@ -236,195 +289,266 @@
 
+for marginal, model_label in zip(dev_marginals_dep_df.columns, model_names):
+    fpr, tpr, threshold = roc_curve(dev_data_labels, dev_marginals_dep_df[marginal])
+    area = auc(fpr, tpr)
+    plt.plot(fpr, tpr, label=model_label+" (AUC = {:0.2f})".format(area))
+    
+plt.title("ROC of DEP Generative Models")
+plt.xlabel("FPR")
+plt.ylabel("TPR")
+plt.legend()
 
-# ## Comparison Between The Amount of Label Functions
 
-# In[31]:
+# In[30]:
 
 
-gen_model_history_df = pd.read_csv("data/disease_gene/disease_associates_gene/gen_model_marginals_history.csv")
+plt.figure(figsize=(10,6))
+positive_class = dev_data_df.curated_dsh.values.sum()/dev_data_df.shape[0]
+plt.plot([0,1], [positive_class, positive_class], color='grey', 
+         linestyle='--', label='Baseline (AUC = {:0.2f})'.format(positive_class))
 
+for marginal, model_label in zip(dev_marginals_indep_df.columns, model_names):
+    precision, recall, threshold = precision_recall_curve(dev_data_labels, dev_marginals_indep_df[marginal])
+    area = auc(recall, precision)
+    plt.plot(recall, precision, label=model_label+" (AUC = {:0.2f})".format(area))
+    
+plt.title("Precision Recall Curve of INDEP Generative Models")
+plt.xlabel("Recall")
+plt.ylabel("Precision")
+plt.legend()
 
-# In[ ]:
 
+# In[31]:
 
-gen_model_history_df['lfs_26_50k'] = L_dev_ci_marginals
-gen_model_history_df.head(2)
 
+L_dev[:, lfs_columns[-1]].lf_stats(session, dev_data_labels, test_df.query("model=='CI'")["Learned Acc."])
 
-# In[32]:
 
+# ## Comparison Between The Amount of Label Functions
 
-positive_class = dev_data_df.curated_dsh.values.sum()/dev_data_df.shape[0]
-plt.plot([0,1], [positive_class, positive_class], color='grey', 
-         linestyle='--', label='Baseline (AUC = {:0.2f})'.format(positive_class))
+# In[32]:
 
-for marginal, model_label in zip(gen_model_history_df.columns, ["17 LFs (AUC {:.2f})", "26 LFs 50k (AUC {:.2f})", "26 LFs 250k (AUC {:.2f})"]):
-    precision, recall, threshold = precision_recall_curve(dev_data_labels, gen_model_history_df[marginal])
-    area = auc(recall, precision)
-    plt.plot(recall, precision, label=model_label.format(area))
-plt.title("Precision Recall Curve of Generative Models")
-plt.xlabel("Recall")
-plt.ylabel("Precision")
-plt.legend()
+
+#gen_model_history_df = pd.read_csv("data/gen_model_marginals_history.csv")
 
 
 # In[33]:
 
 
-plt.plot([0,1], [0,1], linestyle='--', color='grey')
-for marginal, model_label in zip(gen_model_history_df.columns, ["17 LFs (AUC {:.2f})", "26 LFs 50k (AUC {:.2f})", "26 LFs 250k (AUC {:.2f})"]):
-    fpr, tpr, threshold = roc_curve(dev_data_labels, gen_model_history_df[marginal])
-    area = auc(fpr, tpr)
-    plt.plot(fpr, tpr, label=model_label.format(area))
-plt.title("ROC Curve of Generative Models")
-plt.xlabel("False Positive Rate")
-plt.ylabel("True Positive Rate")
-plt.legend()
+#gen_model_history_df['lfs_26'] = L_dev_ci_marginals
+#gen_model_history_df.head(2)
 
 
-# In[ ]:
+# In[34]:
```

Comment:
Why do you have this block commented out?

Reply:
That is blocked out because I didn't want to run that section of the notebook for this experiment. Granted I'm sure there is a better way to handle this situation, but it sufficed for now.

Reply:
This is lightly worrisome and, like @dhimmel, makes me wonder if some of the code in these notebooks should really be functions that are imported and called.

Reply:
Ah ok. Don't want to blow up this PR with a lot of changes, so I am going to upload the changes I made (per the above comments) and then open a new PR for some code restructure and housekeeping.